crates/jiff-aws/src/lib.rs

```diff
@@ -0,0 +1,97 @@
+/*!
+This crate provides type convertion for [Jiff](jiff) and
+Aws Smithy Type [DateTime] used
+in the [AWS SDK for Rust](https://aws.amazon.com/sdk-for-rust/).
+
+```
+use jiff_aws::ConvertAwsDateTime;
+use jiff_aws::ConvertJiffTypes;
+use jiff::Timestamp;
+use jiff::Zoned;
+use aws_smithy_types::DateTime;
+
+let datetime = DateTime::from_secs_and_nanos(1627680004,123000000);
+let timestamp: Timestamp = datetime.try_into_timestamp().unwrap();
+assert_eq!( "2021-07-30T21:20:04.123Z".parse::<Timestamp>().unwrap(), timestamp);
+
+let timestamp = Timestamp::try_from_aws(datetime).unwrap();
+assert_eq!( "2021-07-30T21:20:04.123Z".parse::<Timestamp>().unwrap(), timestamp);
+
+let zoned = datetime.try_into_zoned().unwrap();
+assert_eq!( "2021-07-30T21:20:04.123+00:00[UTC]".parse::<Zoned>().unwrap(), zoned);
+
+let zoned = Zoned::try_from_aws(datetime).unwrap();
+assert_eq!( "2021-07-30T21:20:04.123+00:00[UTC]".parse::<Zoned>().unwrap(), zoned);
+
+let other: DateTime = timestamp.into_aws_datetime();
+assert_eq!(datetime, other);
+
+let other = DateTime::from_timestamp(timestamp);
+assert_eq!(datetime, other);
+
+let other: DateTime = zoned.clone().into_aws_datetime();
+assert_eq!(datetime, other);
+
+let other = DateTime::from_zoned(zoned);
+assert_eq!(datetime, other);
+
+```
+*/
+
+use aws_smithy_types::DateTime;
+use jiff::{tz::TimeZone, Timestamp, Zoned};
+pub use traits::{ConvertAwsDateTime, ConvertJiffTypes};
+mod traits;
+
+impl ConvertAwsDateTime for Timestamp {
+    type Error = jiff::Error;
+
+    fn into_aws_datetime(self) -> DateTime {
+        DateTime::from_timestamp(self)
+    }
+
+    fn try_from_aws(value: DateTime) -> Result<Self, Self::Error> {
+        jiff::Timestamp::new(value.secs(), value.subsec_nanos() as i32)
```

Comment:
Smithy times always have a positive value for nanos, and the value always counts forward in time: https://github.com/smithy-lang/smithy-rs/blob/b34250611c130cb71715ae68e8c6fae86c72677f/rust-runtime/aws-smithy-types/src/date_time/mod.rs#L59

However in jiff, the microsecond value may count either forward or backwards in time, and may be negative: https://github.com/BurntSushi/jiff/blob/460a0bc039f1a8c8381c97d4d207ff755157bee1/src/timestamp.rs#L701

So, I think this will give a wrong conversion for timestamps before the Unix epoch.